index.js

```diff
@@ -2,67 +2,43 @@
 const got = require('got');
 const isPlainObj = require('is-plain-obj');
 
-function ghGot(path, opts) {
-	if (typeof path !== 'string') {
-		return Promise.reject(new TypeError(`Expected \`path\` to be a string, got ${typeof path}`));
-	}
-
-	const env = process.env;
-
-	opts = Object.assign({
+const create = () => got.create({
+	options: got.mergeOptions(got.defaults.options, {
 		json: true,
-		token: env.GITHUB_TOKEN,
-		endpoint: env.GITHUB_ENDPOINT ? env.GITHUB_ENDPOINT : 'https://api.github.com/'
-	}, opts);
-
-	opts.headers = Object.assign({
-		accept: 'application/vnd.github.v3+json',
-		'user-agent': 'https://github.com/sindresorhus/gh-got'
-	}, opts.headers);
-
-	if (opts.token) {
-		opts.headers.authorization = `token ${opts.token}`;
-	}
-
-	// https://developer.github.com/v3/#http-verbs
-	if (opts.method && opts.method.toLowerCase() === 'put' && !opts.body) {
-		opts.headers['content-length'] = 0;
-	}
-
-	const url = /^https?/.test(path) ? path : opts.endpoint.replace(/\/$/, '') + '/' + path.replace(/^\//, '');
+		token: process.env.GITHUB_TOKEN,
+		baseUrl: process.env.GITHUB_ENDPOINT ? process.env.GITHUB_ENDPOINT.replace(/[^/]$/, '$&/') : 'https://api.github.com/',
+		headers: {
+			accept: 'application/vnd.github.v3+json',
+			'user-agent': 'https://github.com/sindresorhus/gh-got'
+		}
+	}),
+	methods: got.defaults.methods,
+	handler: (options, next) => {
+		if (options.token) {
+			options.headers.authorization = `token ${options.token}`;
+		}
 
-	if (opts.stream) {
-		return got.stream(url, opts);
-	}
+		if (options.method && options.method === 'PUT' && !options.body) {
+			options.headers['content-length'] = 0;
+		}
 
-	return got(url, opts).catch(err => {
-		if (err.response && isPlainObj(err.response.body)) {
-			err.name = 'GitHubError';
-			err.message = `${err.response.body.message} (${err.statusCode})`;
+		if (options.stream) {
+			return next(options);
 		}
 
-		throw err;
-	});
-}
+		return next(options).catch(err => {
+			if (err.response && isPlainObj(err.response.body)) {
```

Comment:
Do we really need to check if `body` is a plain object? I think we can just do `if (err.response && err.response.body) {`.